jenkins/release-commit-lock/release-commit-lock.jenkinsfile

```diff
@@ -0,0 +1,168 @@
+                        def buildManifest = readYaml text: buildManifestContent
+                        def existingManifest = readYaml file: "manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml"
+                        def selectedComponents = params.COMPONENTS.isEmpty() ? existingManifest.components : existingManifest.components.findAll { component ->
+                            params.COMPONENTS.split(',').any { it.trim() == component.name }
+                        }
+                        selectedComponents.each { componentName ->
+                            def existingComponent = existingManifest.components.find { it.name == componentName.name }
+                            if (existingComponent) {
+                                def newComponentRef = buildManifest.components.find { it.name == componentName.name }?.ref
+                                if (newComponentRef) {
+                                    existingComponent.ref = newComponentRef
+                                }
+                            }
+                        }
+                        writeYaml file:  "manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml", data: existingManifest, overwrite: true
+                        sh """
+                                    yq eval -i '.' manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml
+                                    sed -i '1s/^/---\\n/' manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml
+                    """
+                    }
+                    updateManifest("opensearch", params.OPENSEARCH_RELEASE_CANDIDATE)
+                    updateManifest("opensearch-dashboards", params.OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE)
+                }
+            }
+        }
+        stage('Update to Recent Commits') {
+            when {
+                expression { params.MANIFEST_LOCK_ACTION == 'Update to Recent Commits' }
+            }
+            steps {
+                script {
+                    def updateManifest = { String productName, String releaseCandidate ->
+                        def existingManifest = readYaml file: "manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml"
+                        def selectedComponents = params.COMPONENTS.isEmpty() ? existingManifest.components : existingManifest.components.findAll { component ->
+                            params.COMPONENTS.split(',').any { it.trim() == component.name }
+                        }
+                        selectedComponents.each { componentName ->
+                            def existingComponent = existingManifest.components.find { it.name == componentName.name }
+                            if (existingComponent) {
+                                def releaseBranch = params.RELEASE_VERSION.split('\\.')[0..1].join('.')
```

Comment:
So this will always be `major.minor` right?

Reply:
So the input is 2.13.0, if will get as 2.13 which is how we are creating the release branch right?